Fix statics in CastorDigiMonitor.

deguio opened the pull request.

Commit:
Fix statics in CastorDigiMonitor.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
JENKINS_TEST_URL

cmsbuild commented:
A new Pull Request was created by @deguio (Federico De Guio) for CMSSW_7_6_X.

Fix statics in CastorDigiMonitor.

It involves the following packages:

DQM/CastorMonitor


@cmsbuild, @danduggan, @vanbesien, @deguio can you please review it and eventually sign? Thanks.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Dr15Jones commented on DQM/CastorMonitor/interface/CastorDigiMonitor.h:
```diff
@@ -44,6 +44,12 @@ class CastorDigiMonitor: public CastorBaseMonitor {
   MonitorElement* hdigisize;
 //  MonitorElement* hBunchOcc;
+
+  int TS_MAX = 10;
```
Looks like this can be made ```constexpr```.

Dr15Jones replied:
I missed it was set in the constructor. However, it could become ```const``` by setting the value before the body of the constructor.

dmitrijus replied:
I am not the author of this, so I want to limit my changes to just removing static variables.

popovvp replied:
TS-MAX can not be const expression as its value is set from python-file

cmsbuild commented:
-1 
Tested at: e303b20e62bf244bb55091e81b11bfd3bf63737c
 When I ran the RelVals I found an error in the following worklfows: 
 4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODreHLT+HARVESTDreHLT.log

----- Begin Fatal Exception 15-Oct-2015 14:40:01 CEST-----------------------

An exception of category 'FatalRootError' occurred while

   [0] Processing run: 194533

   [1] Running path 'dqmoffline_step'

   [2] Calling beginRun for module CastorMonitorModule/'castorOfflineMonitor'

   Additional Info:

      [a] Fatal Root Error: @SUB=TAxis::TAxis::Set

bins must be in increasing order



----- End Fatal Exception -------------------------------------------------

</pre>
9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

----- Begin Fatal Exception 15-Oct-2015 14:49:43 CEST-----------------------

An exception of category 'FatalRootError' occurred while

   [0] Processing run: 1

   [1] Running path 'dqmoffline_step'

   [2] Calling beginRun for module CastorMonitorModule/'castorOfflineMonitor'

   Additional Info:

      [a] Fatal Root Error: @SUB=TAxis::TAxis::Set

bins must be in increasing order



----- End Fatal Exception -------------------------------------------------

</pre>
1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

----- Begin Fatal Exception 15-Oct-2015 14:54:31 CEST-----------------------

An exception of category 'FatalRootError' occurred while

   [0] Processing run: 1

   [1] Running path 'dqmoffline_step'

   [2] Calling beginRun for module CastorMonitorModule/'castorOfflineMonitor'

   Additional Info:

      [a] Fatal Root Error: @SUB=TAxis::TAxis::Set

bins must be in increasing order



----- End Fatal Exception -------------------------------------------------

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

----- Begin Fatal Exception 15-Oct-2015 14:57:04 CEST-----------------------

An exception of category 'FatalRootError' occurred while

   [0] Processing run: 165121

   [1] Running path 'dqmoffline_step'

   [2] Calling beginRun for module CastorMonitorModule/'castorOfflineMonitor'

   Additional Info:

      [a] Fatal Root Error: @SUB=TAxis::TAxis::Set

bins must be in increasing order



----- End Fatal Exception -------------------------------------------------

</pre>
25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

----- Begin Fatal Exception 15-Oct-2015 15:13:01 CEST-----------------------

An exception of category 'FatalRootError' occurred while

   [0] Processing run: 1

   [1] Running path 'dqmoffline_step'

   [2] Calling beginRun for module CastorMonitorModule/'castorOfflineMonitor'

   Additional Info:

      [a] Fatal Root Error: @SUB=TAxis::TAxis::Set

bins must be in increasing order



----- End Fatal Exception -------------------------------------------------

</pre>
50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

----- Begin Fatal Exception 15-Oct-2015 15:13:07 CEST-----------------------

An exception of category 'FatalRootError' occurred while

   [0] Processing run: 1

   [1] Running path 'dqmoffline_step'

   [2] Calling beginRun for module CastorMonitorModule/'castorOfflineMonitor'

   Additional Info:

      [a] Fatal Root Error: @SUB=TAxis::TAxis::Set

bins must be in increasing order



----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11813/8903/summary.html

Commit:
Properly initialize ETower array in DQM/CastorRecHitMonitor.

Previously, the bug  was hidden by the fact statics are initialized to
zero. Moving away from static arrays exposed it.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #11813 was updated. @cmsbuild, @danduggan, @vanbesien, @deguio can you please check and sign again.

deguio commented:
@popovvp
please review the changes in this PR. they are needed to make the code thread safe.
F.

cmsbuild commented:
+1
Tested at: c9889538787c71ee77d1181d767eede57f96f829
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11813/9064/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11813/9064/summary.html

popovvp commented:
[a] Fatal Root Error: @SUB=TAxis::TAxis::Set
bins must be in increasing order

after initializing  ETower[0,1] properly (in the same way as EMTow[] and EhadTow[]) 
the problem will be solved (looks it is done by PR 11904)

deguio commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar
